great_expectations/rule_based_profiler/domain_builder/domain_builder.py

```diff
@@ -21,7 +20,7 @@ class DomainBuilder(ABC):
 
     def __init__(
         self,
-        data_context: DataContext,
+        data_context: "DataContext",  # noqa: F821
         batch_request: Optional[Union[dict, str]] = None,
```

Comment:
@cdkini Per our renewed knowledge, we should use the following throughout:
```
batch_request: Optional[Union[BatchRequest, RuntimeBatchRequest, dict]] = None,
```
in the profiler related classes, except the configuration class (which is strictly `dict` or `None`).  Could you please affect this change?  Thanks!

Reply:
done!